src/v/pandaproxy/rest/proxy.cc

```diff
@@ -22,51 +29,83 @@ namespace pandaproxy::rest {
 
 using server = ctx_server<proxy>;
 
+template<typename Handler>
+auto wrap(Handler h) {
+    return [h{std::move(h)}](
+             server::request_t rq,
+             server::reply_t rp) -> ss::future<server::reply_t> {
+        rq.authn_method = rq.service().config().authn_method(
+          rq.req->get_listener_idx());
+
+        if (rq.authn_method == config::rest_authn_method::http_basic) {
+            // Will throw 400 & 401 if auth fails
+            auto auth_result = rq.service().authenticator().authenticate(
+              *rq.req);
+            // Will throw 403 if user enabled HTTP Basic Auth but
+            // did not give the authorization header.
```

Comment:
I think I'm always getting status 400 for both of these.

Reply:
How are you testing and how are you receiving that error?
I do something like:
1. `rpk acl user create admin --password <secret>`
2. `curl --header "Authorization: Basic <base64 string here>" http://127.0.0.1:8082/brokers` where the credentials are "<username>:<password>" by that whole string is base64 encoded (See the colon between name and pass)

Reply:
> how are you receiving that error?
@BenPope Let me clarify. Do you see the error from curl or in RP logs or somewhere else?

Reply:
Forbidden (no user) - should be 403
```
curl  "http://127.0.0.1:18082/brokers" -v
*   Trying 127.0.0.1:18082...
* Connected to 127.0.0.1 (127.0.0.1) port 18082 (#0)
> GET /brokers HTTP/1.1
> Host: 127.0.0.1:18082
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Content-Length: 71
< Server: Seastar httpd
< Date: Thu, 29 Sep 2022 21:24:29 GMT
< Content-Type: application/json
< 
* Connection #0 to host 127.0.0.1 left intact
{"error_code":40002,"message":"Forbidden (authentication is required)"}
```
Unauthorized (wrong password) - should be 401
```
curl -u "admin:wrong" "http://127.0.0.1:18082/brokers" -v
*   Trying 127.0.0.1:18082...
* Connected to 127.0.0.1 (127.0.0.1) port 18082 (#0)
* Server auth using Basic with user 'admin'
> GET /brokers HTTP/1.1
> Host: 127.0.0.1:18082
> Authorization: Basic YWRtaW46d3Jvbmc=
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Content-Length: 45
< Server: Seastar httpd
< Date: Thu, 29 Sep 2022 21:25:23 GMT
< Content-Type: application/json
< 
* Connection #0 to host 127.0.0.1 left intact
{"error_code":40002,"message":"Unauthorized"}
```

Reply:
I accidentally pushed the change to your branch, but this is the gist of it: https://github.com/redpanda-data/redpanda/pull/6452/commits/0e07091a598995379b8345d7d8679e959b4ee613 but I guess all 40 codes should be handled explicitly.

Reply:
> Your port number is strange `:18082`

I use non-defaults for a little bit extra manual testing.

Reply:
Looks like `request_auth_result::require_authenticated` is just throwing 401 instead of 403. I'm syncing with John on this before I push.